### DISCUSSION TITLE
Labels when importing from checkmk

### DISCUSSION THREAD

**Discussion by MajaSmereczynska** [kuhn-ruess/cmdbsyncer#33] [Q&A] [answered]
I am using Checkmk Raw Edition 2.2.0p7, Netbox v3.5.1 and CMDB Syncer 3.2.5. I am wondering if there is a way to get the label device_type from checkmk as an attribute and then use it as a condition when exporting it to netbox? What I mean is for example when I have a host called host.com in checkmk that is a virtual machine. Then it has a label cmk/device_type:vm. Now I want it to see in the syncer when I view the host and then when I export it to netbox I want to use it as a condition, like for example if device_type is vm (in syncer) then Set Device Role is 1 (in netbox).

**Comment by Bastian-Kuhn** [comment]
Hi Maja,

yes, now possible for a few minutes with the current git version. Just do an Update, then the checkmk inventorize_hosts command should read all labels starting with cmk/ as attributes into the Syncer. Then you can do exactly what you want.

Update to current git is safe today, is tested in multiple environments. 

Just note with the new Version, there is more caching. So basically, when you change something, hit "Commit Changes" to erase the cache. When an imported object is changed, of course, the Cache updates himself for the host.

**Reply by MajaSmereczynska** [reply]
So those are the labels I have with my host in checkmk:
![image](https://github.com/kuhn-ruess/cmdbsyncer/assets/83644808/7aaf3593-0767-4766-a5d6-f98f29777bdb)
The syncer is showing me the site label but not the other three:
![image](https://github.com/kuhn-ruess/cmdbsyncer/assets/83644808/895315a8-38eb-4f21-acec-4c94ec745f07)

**Reply by MajaSmereczynska** [reply]
Also when I try to run ./cmdbsyncer netbox export_hosts netbox I get these errors:
![image](https://github.com/kuhn-ruess/cmdbsyncer/assets/83644808/95868bfe-ad19-4e28-b1cb-960edeeab65a)

**Reply by Bastian-Kuhn** [reply, marked as answer]
This one should be fixed, I had a typo there it seems, sorry. Can you double-check?

**Reply by Bastian-Kuhn** [reply]
The Invenorize seams not to find that labels; otherwise it would look like this:

<img width="589" alt="grafik" src="https://github.com/kuhn-ruess/cmdbsyncer/assets/899110/df67f27a-f4cc-42a8-a768-30ba20838a07">

Can you please send me the api response via E-Mail to mail@bastian-kuhn.de?

If you use this: https://cmdbsyncer.readthedocs.io/en/latest/basics/lcl_config/ config example, you will see the API Requests and responses which I need. Best in a txt file.

**Reply by MajaSmereczynska** [reply]
Yes, the errors are now fixed, thank you
